test/browser/specs/headless-config-inheritance.test.ts

```diff
@@ -0,0 +1,98 @@
+import { expect, test } from 'vitest'
+import { instances, runBrowserTests } from './utils'
+
+interface HeadlessConfigInheritanceTestCase {
+  name: string
+  rootHeadless?: boolean
+  instanceHeadless?: boolean
+  expectedResolvedInstanceHeadless: boolean
+}
+
+test.each<HeadlessConfigInheritanceTestCase>([
+  {
+    name: 'instance headless is used if root headless is not set',
+    rootHeadless: undefined,
+    instanceHeadless: true,
+    expectedResolvedInstanceHeadless: true,
+  },
+  {
+    name: 'instance headless is used if root headless is not set',
+    rootHeadless: undefined,
+    instanceHeadless: false,
+    expectedResolvedInstanceHeadless: false,
+  },
+  {
+    name: 'root headless is used if instance headless is not set',
+    rootHeadless: true,
+    instanceHeadless: undefined,
+    expectedResolvedInstanceHeadless: true,
+  },
+  {
+    name: 'root headless is used if instance headless is not set',
+    rootHeadless: false,
+    instanceHeadless: undefined,
+    expectedResolvedInstanceHeadless: false,
+  },
+  {
+    name: 'both set to the same value',
+    rootHeadless: true,
+    instanceHeadless: true,
+    expectedResolvedInstanceHeadless: true,
+  },
+  {
+    name: 'both set to the same value',
+    rootHeadless: false,
+    instanceHeadless: false,
+    expectedResolvedInstanceHeadless: false,
+  },
+  {
+    name: 'instance headless overrides root headless',
+    rootHeadless: true,
+    instanceHeadless: false,
+    expectedResolvedInstanceHeadless: false,
```

Comment:
this is not correct within your test. you define headless as a CLI argument, not as a value that will be read from the config

in this case, the value _need_ to be `true`, otherwise it's not possible to override with CLI flags

Reply:
oh ok, will circle back with updated test next week. probably just going to duplicate the `vitest.config.ts` file and specify different paths pointing to `vitest-root-true.config.ts`/`vitest-root-false.config.ts`. unless there is api to override the config during test?

Reply:
there is `runInlinedTests` utility that creates them during the test and then deletes the files

Reply:
thanks for the pointer! i've changed the test to use the `runInlineTests` utility in https://github.com/vitest-dev/vitest/pull/7710/commits/2df36e492fdd844598c1e100b09249cab6e6bcc8.

the test isn't passing `headless` as a CLI argument anymore, so i think it's correct now?